conda_mirror/conda_mirror.py

```diff
@@ -339,45 +354,89 @@ def _list_conda_packages(local_dir):
     return fnmatch.filter(contents, "*.tar.bz2")
 
 
-def _validate_packages(package_repodata, package_directory):
+def _validate_packages(package_repodata, package_directory, num_threads=1):
     """Validate local conda packages.
 
-    NOTE: This is slow.
-    NOTE2: This will remove any packages that are in `package_directory` that
+    NOTE1: This will remove any packages that are in `package_directory` that
            are not in `repodata` and also any packages that fail the package
            validation
+    NOTE2: In concurrent mode (num_threads is not None) this will be hard to
+           kill using CTRL-C.
 
     Parameters
     ----------
     package_repodata : dict
         The contents of repodata.json
     package_directory : str
         Path to the local repo that contains conda packages
+    num_threads : int
+        Number of threads to be used for concurrent validation.  Defaults to
+        `num_threads=1` for non-concurrent mode.  To use all available cores,
+        set `num_threads=0`.
     """
     # validate local conda packages
     local_packages = _list_conda_packages(package_directory)
-    for idx, package in enumerate(sorted(local_packages)):
-        # ensure the packages in this directory are in the upstream
-        # repodata.json
-        try:
-            package_metadata = package_repodata[package]
-        except KeyError:
-            logger.warning("%s is not in the upstream index. Removing...",
-                           package)
-            _remove_package(os.path.join(package_directory, package),
-                            reason="Package is not in the repodata index")
-        else:
-            # validate the integrity of the package, the size of the package and
-            # its hashes
-            logger.info('Validating %s. %s of %s', package, idx,
-                        len(local_packages))
-            _validate(os.path.join(package_directory, package),
-                      md5=package_metadata.get('md5'),
-                      size=package_metadata.get('size'))
+
+    # create argument list (necessary because multiprocessing.Pool.map does not
+    # accept additional args to be passed to the mapped function)
+    val_func_arg_list = [(package, num, package_repodata, package_directory)
```

Comment:
I think you want to pass in len(local_packages) as another argument since it's unlikely that `len(local_packages)` == `len(package_repodata)`